### DISCUSSION TITLE
Trying to get Embark tools/Embark Anywhere working in New DF

### DISCUSSION THREAD

**Discussion by TheTimtam97** [DFHack/dfhack#3275] [Q&A] [answered]
I've been trying to understand how Embark tools allowed for embarking anywhere in the old DF so that I could try and get it working for the new UI. I'm struggling though, because it seems like all it's doing is putting some words onto the screen.

I've gotten the plugin to compile and when I enable the plugin, "embark: " shows up in the bottom left.

What am I missing here? I'm obviously very inexperienced with DFHack/DF/coding in general, but if I'm pointed in the right direction I should be able to figure it out. The UI probably operates completely differently due to the changes in the new version and that's why I can't figure it out

Thank you in advance!

EDIT: My mistake, I got the whole Embark tools UI to work, and everything is able to be enabled without error. I just can't embark

**Comment by TheTimtam97** [comment]
![Screenshot 2023-04-21 195212](https://user-images.githubusercontent.com/60720613/233606137-66995ded-17af-4f51-a4a4-a75849c1323d.png)

**Reply by TaxiService** [reply]
(let me preface this by saying that i know next to nothing about how any of this works.)

See how `s: Enabled: ...` has the red `s` before it? That usually means that if you press `s`, you'll do the stuff after the colon.

now, why is it that `: Embark!` does not have such a hotkey? 
what i'm asking is, could it be that you can't embark because, somehow, that hotkey is missing?

I don't really know though; that `: Embark!` thing could just be a text label that was meant to visually overwrite some other text in .47.50... and may just not be functional. :thinking:

**Reply by myk002** [reply, marked as answer]
Most of the functionality of embark tools does not need to be ported since it just won't be useful in the new UI. For example, vanilla provides the Sand indicator now. The embark anywhere bit is the only part that needs work.

Rumrusher actually figured the required logic out, but is having trouble writing the UI part.

scr.warn_mm_startx = scr.neighbor_hover_mm_sx
scr.warn_mm_endx = scr.neighbor_hover_mm_ex
scr.warn_mm_starty = scr.neighbor_hover_mm_sy
scr.warn_mm_endy = scr.neighbor_hover_mm_ey
scr.in_embark_salt=true

https://github.com/DFHack/scripts/pull/638

The old embark tools UI no longer fits in with the DF interface, and I suggest you work with Rumrusher to produce a script with the better UX.

**Reply by lethosor** [reply]
> I don't really know though; that `: Embark!` thing could just be a text label that was meant to visually overwrite some other text in .47.50... and may just not be functional.

That's exactly it (well, 0.47.05, but close enough, haha). The "Embark" text was greyed out by DF when an invalid region by DF's standards was selected, so the plugin drew over it in the white text to indicate that the option was enabled when embark-anywhere was active.
